hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/MkdirOperation.java

```diff
@@ -124,7 +132,32 @@ public Boolean execute() throws IOException {
       return true;
     }
 
-    // Walk path to root, ensuring closest ancestor is a directory, not file
+    // if performance creation mode is set, no need to check
```

Comment:
how about on L116 we only do a HEAD check for the path without /, (maybe need new callback), so no LIST probe for a dir via HEAD/LIST
```
S3AFileStatus fileStatus = performanceCreation
    ? probePathStatusOrNull(dir, StatusProbeEnum.Head) 
    ? getPathStatusExpectingDir(dir);
```

Reply:
Sounds reasonable, i was curious about whether we need full probe for magic, i think yes we can make it much performant.